src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/BigQueryDynamicDestinations.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.changestreams.sample.bigquery.changelog.fullschema;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.changestreams.sample.bigquery.changelog.fullschema.model.SpannerTable;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.teleport.v2.transforms.BigQueryConverters;
+import org.apache.beam.sdk.io.gcp.bigquery.DynamicDestinations;
+import org.apache.beam.sdk.io.gcp.bigquery.TableDestination;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.ValueInSingleWindow;
+
+import java.util.List;
+import java.util.Map;
+
+public class BigQueryDynamicDestinations extends DynamicDestinations<TableRow, KV<TableId, TableRow>> {
+
+  private final Map<String, SpannerTable> spannerTableByName;
+  private final String bigQueryProject, bigQueryDataset, bigQueryTableTemplate;
+
+  BigQueryDynamicDestinations(
+    String spannerProject,
+    String spannerInstance,
+    String spannerDatabase,
+    String spannerHost,
+    String changeStreamName,
+    String bigQueryProject,
+    String bigQueryDataset,
+    String bigQueryTableTemplate) {
+    final DatabaseClient spannerDatabaseClient =
+      SpannerOptions.newBuilder()
+        .setHost(spannerHost)
+        .setProjectId(spannerProject)
+        .build()
+        .getService()
```

Comment:
`getService` returns an object that should be closed on termination

Reply:
Closed.